flask/cli.py

```diff
@@ -539,7 +561,9 @@ def load_dotenv():
 
         dotenv.load_dotenv(path)
```

Comment:
I think before calling `dotenv.load_dotenv()` here you need to make sure the path found by dotenv did not go above `top_dir`. The `top_dir` does not appear to be enforced as it is. For example, with a `.env` file in examples/flaskr and `top_dir=examples/flaskr/flaskr`, the `.env` file is imported anyway (note that I added a print right above `dotenv.load_dotenv()` to help me test this):

```
>>> from flask.cli import load_dotenv
>>> load_dotenv(top_dir='/home/miguel/flask/examples/flaskr/flaskr')
 * Loading environment variables from "/home/miguel/flask/examples/flaskr/.env"
```